src/test/java/org/jabref/model/entry/EntryLinkListTest.java

```diff
@@ -42,4 +67,102 @@ public void givenNullFieldValueAndDatabaseWhenParsingThenExpectLinksIsEmpty() {
         links = EntryLinkList.parse(null, database);
         assertTrue(links.isEmpty());
     }
+
+    @Test
+    public void givenTargetAndSourceWhenSourceCrossrefTargetThenSourceCrossrefsTarget() {
+        target = BibEntryBuild.ing().withId("target").withCiteKey("target").now();
+        source = BibEntryBuild.ing().withId("source").withCiteKey("source").crossref(target).now();
+        assertSourceCrossrefsTarget();
+    }
+
+    @Test
+    public void givenCrossrefBibWhenImportingCrossBibThenSourceCrossrefsTarget() throws IOException {
+        target = crossrefBib.getEntryByKeyOrNew("DBLP:conf/wicsa/2015");
+        source = crossrefBib.getEntryByKeyOrNew("DBLP:conf/wicsa/ZimmermannWKG15");
+        assertSourceCrossrefsTarget();
+    }
+
+    private void assertSourceCrossrefsTarget() {
+        boolean crossrefs = false;
+        Optional<String> sourceCrossref = source.getField(FieldName.CROSSREF);
+        Optional<String> targetCiteKey = target.getCiteKeyOptional();
+        if (sourceCrossref.isPresent()
+                && targetCiteKey.isPresent()) {
+            String actualcrossrefValue = sourceCrossref.get();
+            String expectedCrossrefValue = targetCiteKey.get();
+            crossrefs = actualcrossrefValue.contains(expectedCrossrefValue);
+        }
+        assertTrue(crossrefs);
```

Comment:
Are you aware that you can directly call assertEquals on two optionals?
https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#equals-java.lang.Object-